src/avt/MIR/Youngs/CMakeLists.txt

```diff
@@ -7,22 +7,35 @@
 #
 #****************************************************************************
 
-set(YOUNGS_SOURCES
-    CellInterface.C
-    CellInterface2D.C
-    CellInterface3D.C
-    vtkCQS.C
-    vtkDataSetGradient.C
-    vtkYoungsMaterialInterface.C
-    YoungsMIR.C)
+visit_append_list(
+    NAME  avtmir_ser_SOURCES
+    ITEMS ${CMAKE_CURRENT_SOURCE_DIR}/CellInterface.C
+          ${CMAKE_CURRENT_SOURCE_DIR}/CellInterface2D.C
+          ${CMAKE_CURRENT_SOURCE_DIR}/CellInterface3D.C
+          ${CMAKE_CURRENT_SOURCE_DIR}/vtkCQS.C
+          ${CMAKE_CURRENT_SOURCE_DIR}/vtkDataSetGradient.C
+          ${CMAKE_CURRENT_SOURCE_DIR}/vtkYoungsMaterialInterface.C
+          ${CMAKE_CURRENT_SOURCE_DIR}/YoungsMIR.C)
 
-target_sources(avtmir_ser PRIVATE ${YOUNGS_SOURCES})
-target_include_directories(avtmir_ser PUBLIC .)
+visit_append_list(
+    NAME  avtmir_ser_HEADERS
+    ITEMS ${CMAKE_CURRENT_SOURCE_DIR}/CellInterface.h
+          ${CMAKE_CURRENT_SOURCE_DIR}/CellInterface2D.h
+          ${CMAKE_CURRENT_SOURCE_DIR}/CellInterface3D.h
+          ${CMAKE_CURRENT_SOURCE_DIR}/CellInterfaceCommon.h
+          ${CMAKE_CURRENT_SOURCE_DIR}/CellInterfaceCUDA.h
+          ${CMAKE_CURRENT_SOURCE_DIR}/CellInterfaceMath.h
+          ${CMAKE_CURRENT_SOURCE_DIR}/CellInterfaceTools2D.h
+          ${CMAKE_CURRENT_SOURCE_DIR}/CellInterfaceTools2D_Axis.h
+          ${CMAKE_CURRENT_SOURCE_DIR}/CellInterfaceTools3D.h
+          ${CMAKE_CURRENT_SOURCE_DIR}/CellInterfaceTypes.h
+          ${CMAKE_CURRENT_SOURCE_DIR}/vtkCQS.h
+          ${CMAKE_CURRENT_SOURCE_DIR}/vtkDataSetGradient.h
+          ${CMAKE_CURRENT_SOURCE_DIR}/vtkYoungsMaterialInterface.h
+          ${CMAKE_CURRENT_SOURCE_DIR}/YoungsMIR.h)
```

Comment:
The new approach here is much more verbose than before, with us needing to specify source and header files now. I don't know cmake like you do, so this is probably naive, but why is this so different than before and is there a way for it to be simplified?

Reply:
During our project meeting about this WIP, it was decided we needed to add the header files, too. It enhances CMake tracking of when 'source files' are modified.  It may also help when we begin to utilize some of the blt code tools.

Honestly, the SOURCES and HEADERS get passed together to CMake in an 'add library' call, so the SOURCES and HEADERS lists could be combined into one.  The biggest need for the separate HEADERS arg in the visit_add_library/blt_add_library is for headers-only libraries.

Reply:
I see. I hadn't remembered that from that meeting, forget everything I said!

I think it's neater to separate the headers and sources as you've done but I can see an argument for putting them together too.